src/Induction/WellFounded.agda

```diff
@@ -96,23 +96,22 @@ module All {_<_ : Rel A r} (wf : WellFounded _<_) ℓ where
 module FixPoint
   {_<_ : Rel A r} (wf : WellFounded _<_)
   (P : Pred A ℓ) (f : WfRec _<_ P ⊆′ P)
-  (f-ext : (x : A) {IH IH′ : WfRec _<_ P x} → (∀ {y} y<x → IH y y<x ≡ IH′ y y<x) → f x IH ≡ f x IH′)
+  (f-ext : (x : A) {IH IH′ : WfRec _<_ P x} → (∀ {y} y<x → IH {y} y<x ≡ IH′ y<x) → f x IH ≡ f x IH′)
   where
 
   some-wfRec-irrelevant : ∀ x → (q q′ : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q′
-  some-wfRec-irrelevant = All.wfRec wf _
+  some-wfRec-irrelevant = All.wfRec wf
                                    (λ x → (q q′ : Acc _<_ x) → Some.wfRec P f x q ≡ Some.wfRec P f x q′)
-                                   (λ { x IH (acc rs) (acc rs′) → f-ext x (λ y<x → IH _ y<x (rs _ y<x) (rs′ _ y<x)) })
+                                   (λ { x IH (acc rs) (acc rs′) → f-ext x (λ y<x → IH y<x (rs y<x) (rs′ y<x)) })
 
-  open All wf ℓ
-  wfRecBuilder-wfRec : ∀ {x y} y<x → wfRecBuilder P f x y y<x ≡ wfRec P f y
-  wfRecBuilder-wfRec {x} {y} y<x with wf x
-  ... | acc rs = some-wfRec-irrelevant y (rs y y<x) (wf y)
+  open All wf {ℓ}
+  wfRecBuilder-wfRec : ∀ {x y} y<x → wfRecBuilder P f x y<x ≡ wfRec P f y
+  wfRecBuilder-wfRec {x} {y} y<x with acc rs ← wf x
+   = some-wfRec-irrelevant y (rs y<x) (wf y)
 
-  unfold-wfRec : ∀ {x} → wfRec P f x ≡ f x (λ y _ → wfRec P f y)
+  unfold-wfRec : ∀ {x} → wfRec P f x ≡ f x (λ {y} _ → wfRec P f y)
```

Comment:
And here: artefact of type of `wfRec`, but that I think does need to retain its argument. Properties of it do not, however!